### PR TITLE
TCP: server, client, shell script.

### PR DESCRIPTION
**Pull request by pontuslae**
Server that runs on raspberry pi.
Client that runs on the android application.
Shell script that runs on the raspberry pi.
This only lacks the descripion on how the service was initialized.

### CODE REVIEW

File: tcpclient/Client.java [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+	 *
+	 * @return the first command in the queue.
+	 */
+	public String read(){
+
+		String s;
+		try {
+			in = new BufferedReader(new InputStreamReader(socket.getInputStream())); // I hate Java.
+			s = in.readLine();
+			return "";
```
Comment (Sebastian-Fransson) [inline tcpclient/Client.java]: Return the 's' string instead of an empty one?

**Review by Sebastian-Fransson — DISMISSED** [review]

File: tcpclient/Client.java [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+	 */
+    /*
+    public String readAllFormated(){
+        String[] format = readAll();
+        String formatted = "";
+        for (String f: format)
+            formatted += f + ", ";
+        return formatted.substring(0, formatted.length()-3);
+    }
+*/
```
Comment (timjon) [inline tcpclient/Client.java]: Why is this comment here

Comment (pontuslae) [reply tcpclient/Client.java]: It's for a functionality not currently used after I altered the Android application to not require these.

File: tcpclient/Client.java [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+	public void run() {}
+
+	static void p(String s){
+		System.out.println(s);
+	}
+}
+
+class Transmitter extends BaseSocket implements Runnable {
+
+	Transmitter(String host, int port) {
```
Comment (timjon) [inline tcpclient/Client.java]: What does this do

File: tcpclient/Client.java [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+	}
+
+	/**
+	 *
+	 * @return the first command in the queue.
+	 */
+	public String read(){
+
+		try {
+			in = new BufferedReader(new InputStreamReader(socket.getInputStream())); // I hate Java.
```
Comment (IsabelleTornqvist) [inline tcpclient/Client.java]: Should probably remove this comment

File: tcpclient/Client.java (hunk 3 above)
Comment (timjon) [inline tcpclient/Client.java]: Needs a comment on what this does

File: tcpclient/Client.java [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+	BufferedReader in;
+
+	// A queue is used to handle all input commands so they go in the proper order and are not lost.
+	Queue<String> input = new PriorityQueue<>();
+	Queue<String> output = new PriorityQueue<>(); // TODO: 06/04/2017 make a read function for this.
+	Socket socket;
+
+	// Macro for add.
+	public void write(String s){
+		// TODO: 06/03/2017 Add command vertification here. Preferably O(1).
```
Comment (timjon) [inline tcpclient/Client.java]: Is this something that needs to be done or can it be removed?

File: tcpclient/Client.java [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+			return ""; // Only occurs if an exception is thrown.
+		}
+	}
+    
+	BaseSocket(String host, int port) {
+		this.host = host;
+		this.port = port;
+	}
+
+	public void run() {}
```
Comment (timjon) [inline tcpclient/Client.java]: Remove this?

File: tcpclient/receiver.py [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+
+
+class Receiver:
+
+    receiver = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+    host = None
+    port = None
+    connection = False
+    address = None
+    msg = None
```
Comment (timjon) [inline tcpclient/receiver.py]: Not really needed but maybe a comment for line 17-22 for clarity

**Review by timjon — CHANGES_REQUESTED** [review]
The last things that needs to be checked

**Review by timjon — APPROVED** [review]

**Pull request merged**